File: AbyssalCreatures/src/main/java/com/offz/spigot/abyssalcreatures/Mobs/Flying/Madokajack.java
package com.offz.spigot.abyssalcreatures.Mobs.Flying;

import com.offz.spigot.mobzy.mobs.behaviours.HitBehaviour;
import com.offz.spigot.mobzy.mobs.types.FlyingMob;
import com.offz.spigot.mobzy.pathfinders.flying.PathfinderGoalDiveOnTargetAttack;
import net.minecraft.server.v1_13_R2.World;

public class Madokajack extends FlyingMob implements HitBehaviour {
    public Madokajack(World world) {
        super(world, "Madokajack");
        setSize(4.5f, 3f);
    }

    @Override
    public void createPathfinders() {
        super.createPathfinders();
        goalSelector.a(2, new PathfinderGoalDiveOnTargetAttack(this, -0.3, 6, 10, 8, 2, 0.6, 30));
    }
}

File: AbyssalCreatures/src/main/java/com/offz/spigot/abyssalcreatures/mobs/flying/Madokajack.kt
package com.offz.spigot.abyssalcreatures.mobs.flying

import com.offz.spigot.mobzy.mobs.behaviours.HitBehaviour
import com.offz.spigot.mobzy.mobs.types.FlyingMob
import com.offz.spigot.mobzy.pathfinders.flying.PathfinderGoalDiveOnTargetAttack
import net.minecraft.server.v1_15_R1.World

class Madokajack(world: World?) : FlyingMob(world, "Madokajack"), HitBehaviour {
    override fun createPathfinders() {
        super.createPathfinders()
        addPathfinderGoal(2, PathfinderGoalDiveOnTargetAttack(
                this,
                minHeight = 6.0,
                maxHeight = 10.0,
                startDiveDistance = 14.0,
                bashVelMultiplier = 1.0
        ))
    }
}